src/core/file_format/psf.cpp

```diff
@@ -0,0 +1,59 @@
+// SPDX-FileCopyrightText: Copyright 2024 shadPS4 Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include <cstring>
+#include <fstream>
+#include <iostream>
+
+#include "common/io_file.h"
+#include "psf.h"
+
+PSF::PSF() = default;
+
+PSF::~PSF() = default;
+
+bool PSF::open(const std::string& filepath) {
+    Common::FS::IOFile file(filepath, Common::FS::FileAccessMode::Read);
+    if (!file.IsOpen()) {
+        return false;
+    }
+
+    const u64 psfSize = file.GetSize();
+    psf.resize(psfSize);
+    file.Seek(0);
+    file.ReadRaw<u8>(&psf[0], psfSize);
```

Comment:
```suggestion
    file.Read(psf);
```
Also rename psfSize to psf_size